.github/workflows/validate-new-issue.yml

```diff
@@ -0,0 +1,61 @@
+name: Validate new issue
+on:
+  issues:
+    types: ['opened']
+jobs:
+  validate-new-issue:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: "Validate issue against templates"
+      shell: bash
+      env:
+        GITHUB_TOKEN: ${{ github.token }}
+      run: |
+        echo "Validating issue #${{ github.event.issue.number }}."
+
+        # Trust users who belong to the getsentry org.
+        if gh api "https://api.github.com/orgs/getsentry/members/${{ github.actor }}" >/dev/null 2>&1; then
+          echo "Skipping validation, because ${{ github.actor }} is a member of the getsentry org."
+          exit 0
+        else
+          echo "${{ github.actor }} is not a member of the getsentry org. 🧐"
+        fi
+
+        # Look for a template where all the headings are also in this issue.
+        # - extra headings in the issue are fine
+        # - order doesn't matter
+        # - case-sensitive tho
+        # - can't post a template unchanged (ignoring whitespace)
+        function extract-headings { { grep '^#' "$1" || echo -n ''; } | sort; }
+        jq -r .issue.body "$GITHUB_EVENT_PATH" > issue
+        extract-headings <(cat issue) > headings-in-issue
+        for template in $(ls .github/ISSUE_TEMPLATE/*.md 2> /dev/null); do
+          # Strip front matter. https://stackoverflow.com/a/29292490/14946704
+          sed -i'' '1{/^---$/!q;};1,/^---$/d' "$template"
+          extract-headings "$template" > headings-in-template
+          echo -n "$(basename $template)? "
+          if [ ! -s headings-in-template ]; then
```

Comment:
nit: if you're using bash you may as well use builtin `[[`, not `[` which is usually `/usr/bin/[`.

Reply:
I'm vaguely familiar with this. Why would I prefer one to the other?